orc/src/main/java/org/apache/iceberg/orc/EstimateOrcAveWidthVisitor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.orc;
+
+import java.util.List;
+import java.util.Optional;
+import org.apache.orc.TypeDescription;
+
+public class EstimateOrcAveWidthVisitor extends OrcSchemaVisitor<Integer> {
+
+  @Override
+  public Integer record(TypeDescription record, List<String> names, List<Integer> fields) {
+    return fields.stream().reduce(Integer::sum).orElse(0);
+  }
+
+  @Override
+  public Integer list(TypeDescription array, Integer elementResult) {
+    return elementResult;
+  }
+
+  @Override
+  public Integer map(TypeDescription map, Integer keyWidth, Integer valueWidth) {
+    return keyWidth + valueWidth;
+  }
+
+  @Override
+  public Integer primitive(TypeDescription primitive) {
+    Optional<Integer> icebergIdOpt = ORCSchemaUtil.icebergID(primitive);
+
+    if (!icebergIdOpt.isPresent()) {
+      return 0;
+    }
+
+    switch (primitive.getCategory()) {
```

Comment:
I think we need to align with the approach to estimate the avg width for each data type.   I think the basic rule is: we  need to read the GenericOrcWriters to see how those data types are encouded into the ORC column vector. That is the occupied in-memory byte size without any columnar compression.

Reply:
The corresponding relationship is as follows:
```
Boolean	->	LongColumnVector
Byte	->	LongColumnVector
Short	->	LongColumnVector
INT	->	LongColumnVector
LONG 	->	LongColumnVector 	
FLOAT	->	DoubleColumnVector
DOUBLE	->	DoubleColumnVector
DATE	->	LongColumnVector
TIMESTAMP	->	TimestampColumnVector
BINARY	->	BytesColumnVector
STRING	->	BytesColumnVector
DECIMAL	->	Decimal18Writer or Decimal38Writer
```
The byte estimation corresponds:
```
LongColumnVector	->	8 byte
DoubleColumnVector	->	8 byte
TimestampColumnVector	->	12 byte
Decimal18Writer/Decimal38Writer	->	(precision + 4) / 2 byte
BytesColumnVector	->	128 byte
```
How about this?

Reply:
The estimated byte size of decimal is just:  precision + 2.  Just as I said in another comment,  each digit will occupy just one byte.  and in fact,  the `BigDecimal`'s unscaled value is usually a `BigInteger`, and the `BigInteger` will just encode each digit into a byte.